packages/application-shell/src/components/authenticated/auth-callback.tsx

```diff
@@ -0,0 +1,118 @@
+import type { ApplicationWindow } from '@commercetools-frontend/constants';
+import type { AuthorizeSessionState } from './types';
+
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { decode } from 'qss';
+import jwtDecode from 'jwt-decode';
+import Card from '@commercetools-uikit/card';
+import Constraints from '@commercetools-uikit/constraints';
+import { ContentNotification } from '@commercetools-uikit/notifications';
+import Spacings from '@commercetools-uikit/spacings';
+import Text from '@commercetools-uikit/text';
+import { STORAGE_KEYS, LOGIN_STRATEGY_OIDC } from '../../constants';
+import Redirector from '../redirector';
+
+declare let window: ApplicationWindow;
+
+type AuthorizeCallbackFragments = { sessionToken?: string; state: string };
+type SessionToken = { nonce: string };
+
+const removeSSOSessionState = (key: string) =>
+  window.sessionStorage.removeItem(key);
+const getSSOSessionState = (key: string): AuthorizeSessionState | null => {
+  const unparsedSessionState = window.sessionStorage.getItem(key);
+  if (unparsedSessionState) {
+    try {
+      const parsedSessionState = JSON.parse(
+        unparsedSessionState
+      ) as AuthorizeSessionState;
+
+      removeSSOSessionState(key);
+
+      return parsedSessionState;
+    } catch (error) {
+      removeSSOSessionState(key);
+
+      if (process.env.NODE_ENV !== 'production') {
+        // eslint-disable-next-line no-console
+        console.warn(
+          `Cannot parse session state for "${key}".\n${unparsedSessionState}`
+        );
+      }
+    }
+  }
+  return null;
+};
+
+const AuthCallback = () => {
```

Comment:
This is the component that handles the auth callback redirect (after login). Most of the things here are very similar to what we also do for the SSO callback.